src/ln/channelmanager.rs

```diff
@@ -8060,4 +8057,412 @@ mod tests {
 		let spend_tx = check_static_output!(events, nodes, 0, 0, 2, 1);
 		check_spends!(spend_tx, closing_tx);
 	}
+
+	fn run_onion_failure_test<F1,F2>(_name: &str, test_case: u8, nodes: &Vec<Node>, route: &Route, payment_hash: &[u8; 32], callback_msg: F1, callback_node: F2, expected_retryable: bool, expected_error_code: Option<u16>, expected_channel_update: Option<HTLCFailChannelUpdate>)
+		where F1: for <'a> FnMut(&'a mut msgs::UpdateAddHTLC),
+					F2: FnMut(),
+	{
+		run_onion_failure_test_with_fail_intercept(_name, test_case, nodes, route, payment_hash, callback_msg, |_|{}, callback_node, expected_retryable, expected_error_code, expected_channel_update);
+	}
+
+	// test_case
```

Comment:
It'd be nice to get a bit more of a description of exactly what this does and what the callbacks mean as its not super trivial to read.